src/main/java/io/confluent/connect/elasticsearch/ClientConfigCallbackHandler.java

```diff
@@ -151,6 +191,64 @@ private PoolingNHttpClientConnectionManager configureConnectionManager() {
     }
   }
 
+  /**
+   * Configures the client to use Kerberos authentication. Overrides any proxy or basic auth
+   * credentials.
+   *
+   * @param builder the HttpAsyncClientBuilder to configure
+   * @return the configured builder
+   */
+  private HttpAsyncClientBuilder configureKerberos(HttpAsyncClientBuilder builder) {
+    GSSManager gssManager = GSSManager.getInstance();
+    Lookup<AuthSchemeProvider> authSchemeRegistry =
+        RegistryBuilder.<AuthSchemeProvider>create()
+            .register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory())
+            .build();
+    builder.setDefaultAuthSchemeRegistry(authSchemeRegistry);
+
+    try {
+      LoginContext loginContext = login();
+      GSSCredential credential = Subject.doAs(
+          loginContext.getSubject(),
+          (PrivilegedExceptionAction<GSSCredential>) ()
+              -> gssManager.createCredential(
+              null,
+              GSSCredential.DEFAULT_LIFETIME,
+              SPNEGO_OID,
+              GSSCredential.INITIATE_ONLY
+          )
+      );
+      CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+      credentialsProvider.setCredentials(
+          new AuthScope(
+              AuthScope.ANY_HOST,
+              AuthScope.ANY_PORT,
+              AuthScope.ANY_REALM,
+              AuthSchemes.SPNEGO
+          ),
+          new KerberosCredentials(credential)
+      );
+      builder.setDefaultCredentialsProvider(credentialsProvider);
+    } catch (PrivilegedActionException e) {
+      throw new ConnectException(e);
+    }
+
+    if (!config.isSslEnabled()) {
+      try {
+        SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
+        sslContextBuilder.loadTrustMaterial(null, new TrustAllStrategy());
+        builder.setSSLContext(sslContextBuilder.build());
```

Comment:
the default SSL Context is to have this on in the builder? Wondering why we need to create a null builder when SSL is disabled